src/NBitcoin/ConsensusFactory.cs

```diff
@@ -99,25 +99,22 @@ private bool IsAssignable<T>(TypeInfo type, ConcurrentDictionary<Type, bool> cac
         /// <typeparam name="T">The generic type to resolve.</typeparam>
         /// <param name="result">If the type is known it will be initialized.</param>
         /// <returns><c>true</c> if it is known.</returns>
-        public virtual bool TryCreateNew<T>(out T result) where T : IBitcoinSerializable
+        public virtual T TryCreateNew<T>() where T : IBitcoinSerializable
```

Comment:
Why this change I wonder?

Reply:
Hi Dan! OK so when I made `BlockHeader`'s constructor internal, the `Clone` method didn't like it because it was looking for a public parameter-less constructor. This was due to the `new()` generic constraint (old code):

```public static T Clone<T>(this T serializable, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, Network network = null) where T : IBitcoinSerializable, new()```

Because of that change we now need to create the default instance via the `Activator`.